pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -425,6 +425,13 @@ public void removeProducer(Producer producer) {
 
         replicators.forEach((cluster, replicator) -> futures.add(replicator.disconnect()));
         producers.values().forEach(producer -> futures.add(producer.disconnect()));
+        if (topicPublishRateLimiter instanceof AutoCloseable) {
```

Comment:
why can't we make every instance of topicPublishRateLimiter  implement `AutoCloseable` ?

Reply:
yes, I guess it makes sense in this case since PublishRateLimiter is an internal interface and it's fine to change it.

Reply:
@eolivelli this is now resolved. PTAL